core/ops.rs

```diff
@@ -120,6 +123,22 @@ pub fn to_op_result<R: Serialize + 'static>(
   }
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum OpMetricsEvent {
+  /// Entered an op.
+  Enter,
+  /// Left an op synchronously.
+  Leave,
+  /// Left an op asynchronously.
+  LeaveAsync,
+  /// Left an op synchronously with an exception.
+  Exception,
+  /// Left an op asynchronously with an exception.
+  ExceptionAsync,
+}
```

Comment:
Bikeshedding time: `Completed(Async)` and `Error(Async)`?

Reply:
`Started`/`Completed`?

Reply:
Previously we used "dispatched" and "completed". That works too

Reply:
I went with Dispatched/Completed.